lib/api/bucketGetEncryption.js

```diff
@@ -0,0 +1,93 @@
+const { errors, s3middleware } = require('arsenal');
+
+const bucketShield = require('./apiUtils/bucket/bucketShield');
+const { isBucketAuthorized } =
+    require('./apiUtils/authorization/permissionChecks');
+const metadata = require('../metadata/wrapper');
+const { pushMetric } = require('../utapi/utilities');
+const escapeForXml = s3middleware.escapeForXml;
+const collectCorsHeaders = require('../utilities/collectCorsHeaders');
+
+const requestType = 'bucketGetEncryption';
+
+/**
+ * Bucket Get Encryption - Get bucket SSE configuration
+ * @param {AuthInfo} authInfo - Instance of AuthInfo class with requester's info
+ * @param {object} request - http request object
+ * @param {object} log - Werelogs logger
+ * @param {function} callback - callback to server
+ * @return {undefined}
+ */
+
+function bucketGetEncryption(authInfo, request, log, callback) {
+    const bucketName = request.bucketName;
+    const canonicalID = authInfo.getCanonicalID();
+
+    return metadata.getBucket(bucketName, log, (err, bucket) => {
+        if (err) {
+            log.debug('metadata getbucket failed', { error: err });
+            return callback(err);
+        }
+        if (bucketShield(bucket, requestType)) {
+            return callback(errors.NoSuchBucket);
+        }
+        log.trace('found bucket in metadata', {
+            bucket: bucketName,
+            method: 'bucketGetEncryption',
+        });
+
+        const corsHeaders = collectCorsHeaders(request.headers.origin,
+            request.method, bucket);
+
+        if (!isBucketAuthorized(bucket, requestType, canonicalID, authInfo,
+        log)) {
+            log.debug('access denied for account on bucket', {
+                requestType,
+                method: 'bucketGetEncryption',
+            });
+            return callback(errors.AccessDenied, null, corsHeaders);
+        }
+
+        const sseInfo = bucket.getServerSideEncryption();
+
+        // If sseInfo is present but the `mandatory` flag is not set
+        // then this info was not created using bucketPutEncryption
```

Comment:
Maybe you could mention that it was also not created by the `x-amz-scal-server-side-encryption` header at bucket creation.